### PR TITLE
Fix datasets io with RecoPSFMap

### PR DESCRIPTION
**Pull request by QRemy**
This PR fix the loading of the datasets with RecoPSFMap.
The changes to deal with the lazy loading are not very nice, any better idea @adonath ?

### CODE REVIEW

**Review by registerrier — DISMISSED** [review]
Thanks @QRemy .
I have left two inline comments.

File: gammapy/datasets/map.py [hunk 1]
```diff
@@ -1187,29 +1187,14 @@ def from_hdulist(cls, hdulist, name=None, lazy=False, format="gadf"):
             )
 
         if "EDISP" in hdulist:
-            edisp_map = Map.from_hdulist(hdulist, hdu="edisp", format=format)
-
-            try:
```
[the remark this answers is not in the document]
Comment (QRemy) [reply gammapy/datasets/map.py:1192]: It relies on `IRFMap.from_hdulist` that can handle this case : https://github.com/gammapy/gammapy/blob/73dd5dd98e300349087c3e6a450e228cfc4c5c64/gammapy/irf/core.py#L734-L745

And this is tested by :
https://github.com/gammapy/gammapy/blob/73dd5dd98e300349087c3e6a450e228cfc4c5c64/gammapy/irf/psf/tests/test_map.py#L561-L565

but I added an assert to check explicitly that the `exposure_map` is None

**Review by LauraOlivera — APPROVED** [review]
Thanks a lot @QRemy ! Everything looks good to me, I tested it locally with some HAWC datasets that have a reco PSF with exposure map = `None` and it all worked fine. So I think this is ready to merge :+1:

File: gammapy/datasets/map.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (LauraOlivera) [reply gammapy/datasets/map.py:1192]: this looks OK to me

**Review by registerrier — APPROVED** [review]
Thanks @QRemy !

**Pull request merged**